File: solutions/csharp/resistor-color/1/ResistorColor.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ResistorColor
{
    private static readonly StringComparison Comparison
        = StringComparison.OrdinalIgnoreCase;

    private static IEnumerable<string> Resistors
        => new[] { "black", "brown", "red", "orange", "yellow",
             "green", "blue", "violet", "grey", "white" };

    public static int ColorCode(string color)
        => Resistors
            .ToList()
            .FindIndex(r => r.Equals(color, Comparison));

    public static string[] Colors()
        => Resistors.ToArray();
}

File: solutions/c/resistor-color/1/resistor_color.c
#include "resistor_color.h"

uint16_t color_code(resistor_band_t color)
{
    return (uint16_t)color;
}

resistor_band_t* colors(void)
{
    const int length = WHITE + 1;
    static resistor_band_t colors_values[length];

    for (int i = 0; i < length; i++)
    {
        colors_values[i] = (resistor_band_t)i;
    }

    return colors_values;
}
